docs/007-funding-rate.adoc

```diff
@@ -0,0 +1,96 @@
+= ADR 007 - Funding rate
+Lucas Soriano <lucas@10101.finance>
+:toc:
+:icons: font
+:attributes: 2023-09-06
+
+Status: Proposed
+
+== Decision
+
+The funding rate should be charged through a Lightning payment.
+The funding rate *must* be paid before renewing the contract.
+No funding rate needs to paid for the period of time _before_ the first renewal.
```

Comment:
One important thing we should consider is that funding rate is charged retrospectively and not in advance, e.g. 

- the funding rate should be charged every 8h (00:00, 8:00, 16:00 - aka `funding event`)
- a position which is open at these times should require a funding rate to remain open
- the funding is paid at the `funding event`
- a position which was only opened between two `funding events`, may not be charged, e.g. a position open between 00:01 and 07:59
  - this is ok, because you can't _close_ a position, you can only sell it to someone else in the market who then has the position open past the next funding event. 

The problem I want to solve is the following: 
- you might have multiple positions before a funding event, but only one during the funding event, e.g. a user who constantly buys and sells opened and closed his position multiple times. 
  - If we were to charge the user the funding rate of 8h in advance, we would need to charge him for each trade, 
  - while if we charge him retrospectively at the funding event, we only charge him once

Reply:
> the funding rate should be charged every 8h

I was thinking about this yesterday and this means the user needs to come online every 8 hours. I guess this might be solved by our work on background tasks though. Long expiries (currently set to 7 days, although pretty arbitrary) might not work well with funding rates.

Updating the expiry and charging the funding rate have to happen together because the funding rate is paid precisely to have the right to keep the position open.

> The problem I want to solve is the following:
>
>- you might have multiple positions before a funding event, but only one during the funding event, e.g. a user who constantly buys and sells opened and closed his position multiple times.
>     - If we were to charge the user the funding rate of 8h in advance, we would need to charge him for each trade,
>     - while if we charge him retrospectively at the funding event, we only charge him once

I think the proposed solution does respect this, but we cannot ensure atomicity (for now) if something goes awry.